apps/designer/app/canvas/custom-components/link.tsx

```diff
@@ -0,0 +1,48 @@
+import { useLocation } from "@remix-run/react";
+import type {
+  ElementRef,
+  ComponentProps,
+  ForwardRefExoticComponent,
+  RefAttributes,
+} from "react";
+import { forwardRef } from "react";
+import { preserveSearchBuildParams } from "~/shared/router-utils";
+
+// href starts with http:, mailto:, tel:, etc.
+const isAbsoluteUrl = (href: string) => /^[a-z]+:/i.test(href);
+
+export const preserveBuildParams = (href: string, sourceSearch: string) => {
+  if (isAbsoluteUrl(href)) {
+    return href;
+  }
+
+  const url = new URL(href, "http://localhost");
```

Comment:
I would write that localhost here has no sense and used as fake host just to use URL